pkg/telemetry/telemetry.go

```diff
@@ -32,12 +33,15 @@ type engineImpl struct {
 	uuid [16]byte
 	conn pgxconn.PgxConn
 
+	keeper           housekeeper
+	stopHousekeeping func() error
+
 	stop chan struct{}
 
 	metrics sync.Map
 }
 
-func NewEngine(conn pgxconn.PgxConn, uuid [16]byte) (*engineImpl, error) {
+func NewEngine(conn pgxconn.PgxConn, uuid [16]byte, connStr string) (*engineImpl, error) {
```

Comment:
This might need a bit of rethinking as well. Needing a connection and a connection string to create an engine feels wrong.

Reply:
Suggestions will be helpful. We cannot pass in connStr in Housekeeper() because pg_try_advisory_lock will always be accepted for same session.